clippy_lints/src/methods/map_unwrap_or.rs

```diff
@@ -70,9 +81,29 @@ pub(super) fn lint<'a, 'tcx>(
                 "replace `map({}).unwrap_or({})` with `{}`",
                 map_snippet, unwrap_snippet, suggest
             );
-            span_note_and_lint(cx, OPTION_MAP_UNWRAP_OR, expr.span, msg, expr.span, &note);
+            span_note_and_lint(
+                cx,
+                if is_option {
+                    OPTION_MAP_UNWRAP_OR
+                } else {
+                    RESULT_MAP_UNWRAP_OR
+                },
```

Comment:
This expr could be re-assigned into a variable earlier to avoid duplicating below.

Reply:
good point. will do.